### DISCUSSION TITLE
SocketTimeoutException for Android

### DISCUSSION THREAD

**Discussion by olezvinskyi** [mitmproxy/mitmproxy#6511] [Q&A]
Hi, could you please help me? I don't know if it's related to mitmproxy but you may have some ideas. When I turn on mitmproxy I get SocketTimeoutException exception after some operations in app.  For Charles everything works fine

Presetup
- mitmproxy 10.1.5 
- emulator Android Pixel 5 SDK 30
- Android app: https://firebasestorage.googleapis.com/v0/b/books-around-2fbb6.appspot.com/o/tunein1.apk?alt=media&token=ece1d967-10c1-43dd-9e12-320cf2bd5a19

Steps:
1) Run mitmweb `mitmweb --listen-port 8089 --allow-hosts 'event.atform.tunein.com|54.214.159.221|44.224.15.95|44.239.168.150|44.235.161.248|tunein.adswizz.com|52.51.184.3|3.248.28.79|52.31.181.4|63.34.220.208|34.253.52.213|52.30.151.40|54.247.136.41|52.208.158.104|tunein.deliveryengine.adswizz.com|54.247.136.41|3.248.28.79|34.253.52.213|52.51.184.3|52.31.181.4|63.34.220.208|52.30.151.40|52.208.158.104|reports.radiotime.com|104.17.106.184|104.17.107.184|feed.radiotime.com|104.18.34.155|172.64.153.101|pubads.g.doubleclick.net|172.217.16.2|googleads.g.doubleclick.net|142.250.75.2|securepubads.g.doubleclick.net|216.58.215.66|insight.adsrvr.org|35.71.131.137|3.33.220.150|15.197.193.217|52.223.40.198|opml.radiotime.com|104.17.107.184|104.17.106.184|ad-hls.tunein.com|18.66.233.44|18.66.233.125|18.66.233.94|18.66.233.103|ms4.applovin.com|34.102.162.219|media.admob.com|142.250.147.138|142.250.147.113|142.250.147.139|142.250.147.101|142.250.147.100|142.250.147.102|googletagservices.com|unknown|googlesyndication.com|142.250.203.132|g.doubleclick.net|unknown|googleadservices.com|142.250.203.130|t1.streamguys1.com|75.102.17.41|aax.amazon-adsystem.com|18.244.110.207|sdk.iad-05.braze.com|146.75.118.133|aax-eu.amazon-adsystem.com|52.95.115.196|firebaselogging-pa.googleapis.com|172.217.16.10|142.250.203.138|172.217.16.42|142.250.186.202|216.58.208.202|142.250.203.202|aax-eu.amazon-adsystem.com|52.95.115.196|telemetry.sdk.inmobi.com|20.39.59.149|18.238.192.37|18.238.192.112|18.238.192.73|18.238.192.119'
`
2) Run emulator. I used this tutorial to setup mitmproxy for android https://www.headspin.io/blog/capturing-android-emulator-network-traffic-with-appium
3) Open TuneIn app
4) Open Search 
5) Input in search aa6-s324660
6) Open station "QA Adswizz Instream Test"
7) Station should play but after buffering I get "Stream connection error occured"
In mitmweb I see some requests 

https://gist.github.com/olezvinskyi/a05691eb6c109c9ebe91614e86fbe527
`https://opml.radiotime.com/Report.ashx?c=eventlist&id=s324660&itemToken=BgQEAAEAAQABAAEAUVEAAQQAAA&listenId=1701297924455&orientation=PORTRAIT&audioport=phone_speaker&itemUrlScheme=secure&con=Android%7Cunknown%7C310%7C260&latlon=&provider=ggl&serial=99b57a53-8e41-4a8b-97e0-58d5af77365e&partnerId=xwhZkVKi&render=json&version=33.0&device=PHONE&resolution=0%2C0 HTTP/2.0`

Encoded data :
`event: audio|remoteErrorLog|streamId=e478930100.listenerId=1701297924455.date=2023-11-29 22:45:35.guideId=s324660.itemToken=BgQEAAEAAQABAAEAUVEAAQQAAA.uri=/data/user/0/tunein{DOT}player/app_hls/cache{DOT}m3u8.elapsedStreamTimeMs=1627498.loadDurationMs=10476.bytesLoaded=0.errorMessage=java{DOT}io{DOT}IOException: java{DOT}util{DOT}concurrent{DOT}ExecutionException: java{DOT}net{DOT}SocketTimeoutException: timeout.errorStackTrace=androidx{DOT}media3{DOT}datasource{DOT}HttpDataSource$HttpDataSourceException: java{DOT}io{DOT}IOException: java{DOT}util{DOT}concurrent{DOT}ExecutionException: java{DOT}net{DOT}SocketTimeoutException: timeout`

https://gist.github.com/olezvinskyi/487824b235969dd8f4efa1c7fc6056b8
`POST https://opml.radiotime.com/Report.ashx?c=eventlist&id=s324660&itemToken=BgQEAAEAAQABAAEAUVEAAQQAAA&listenId=1701297924455&orientation=PORTRAIT&audioport=phone_speaker&itemUrlScheme=secure&con=Android%7Cunknown%7C310%7C260&latlon=&provider=ggl&serial=99b57a53-8e41-4a8b-97e0-58d5af77365e&partnerId=xwhZkVKi&render=json&version=33.0&device=PHONE&resolution=0%2C0 HTTP/2.0`

Encoded data :
`event: audio|playerErrorLog|streamId=e478930100.listenerId=1701297924455.date=2023-11-29 22:45:35.guideId=s324660.itemToken=BgQEAAEAAQABAAEAUVEAAQQAAA.url=http://t1/{DOT}streamguys1{DOT}com/althtfree1-stage?aw_0_1st{DOT}playerid=xwhZkVKi&aw_0_1st{DOT}skey=1701297924&aw_0_1st{DOT}bundleId=tunein{DOT}player&listenerid=276afa55-f908-465f-a80b-fd243ade7dde&aw_0_1st{DOT}abtest=&partnerId=xwhZkVKi&aw_0_1st{DOT}stationId=s324660&aw_0_1st{DOT}premium=false&source=TuneIn&us_privacy=1NNY&aw_0_1st{DOT}platform=tunein&aw_0_1st{DOT}ads_partner_alias=mob{DOT}AndroidFree&aw_0_azn{DOT}planguage=en&aw_0_1st{DOT}lotamesegments=%3ball%3b.errorCode=2001.errorName=ERROR_CODE_IO_NETWORK_CONNECTION_FAILED.streamPositionMs=1627503.causeStackTrace=androidx{DOT}media3{DOT}datasource{DOT}HttpDataSource$HttpDataSourceException: java{DOT}io{DOT}IOException: java{DOT}util{DOT}concurrent{DOT}ExecutionException: java{DOT}net{DOT}SocketTimeoutException: timeout`

**Comment by barteo** [comment]
@olezvinskyi Did you find the solution to this problem already? I'm experiencing the same issue.
